### ISSUE TITLE
Having a scene with a node with a " in the name corrupts scene

### ISSUE THREAD

**Issue by Bauxitedev** [godotengine/godot#18448]
**Godot version:**

3.0.2

**OS/device including version:**
KDE Neon

**Issue description:**
If you put a " (quote) in the name of a node, and then save the scene, the scene file is corrupted and cannot be read again by Godot.

**Steps to reproduce:**
1. Create new project with root node.
2. Name it "hello" including quotes
3. Save and close the scene
4. Try to load scene again and be greeted with an error: "Error parsing scene file"

**Comment by Zylann** [comment]
Data injection? :D

**Comment by KidRigger** [comment]
Should remove the quotes while saving?

**Comment by mhilbrunner** [comment]
>Should remove the quotes while saving? Or preserve?

IMO, quotes should either be disallowed *while naming a node* or properly escaped.

**Comment by KidRigger** [comment]
@mhilbrunner Adding escapes while saving will work (added escapes to the saved file, works fine unless I save over it, in which case it breaks again as the escapes are removed)

**Comment by KidRigger** [comment]
Disallowing while naming. Easier. Any other characters to be taken care of?

**Comment by Zylann** [comment]
Note that nodes can be named elsewhere than just the editor. The question is, should it be allowed, and if not, check in `Node.set_name()`.

**Comment by KidRigger** [comment]
Adding the restrictions in set_name() as well. It already handles ":", "@", "/". I'm adding " and "."
Similarly, to the editor, which handles "." and "/", I'm adding ", ":" and "@" for consistency.
Seems right to me. (Union of all forbidden characters in both ways)

**Comment by Zylann** [comment]
Actually I edited my post because I realized it could also slowdown scene loading a little bit, because I think `set_name` is used by the scene loader.

**Comment by KidRigger** [comment]
Yeah, so, what should be done (Well the best case will be for scene_loader to use a different set_name from the one available to users where only the user version does the input vetting. Shouldn't be too difficult but certainly needs more discussion.)

**Comment by KidRigger** [comment]
Apparently this is exactly how it's done. set_name is not used by scene loader. (It uses a nocheck version) 
So I will just add these cases in the change in set_name. 
#18451 takes care of editor, I'll add the code side.
